proposals/12020-model-registry-integration/README.md

```diff
@@ -0,0 +1,432 @@
+    "modelName": "sentiment-classifier",
+    "versionName": "my-model-v1.0.0",
+    "versionID": 42,
+    "modelID": 15,
+    "modelRegistryURL": "https://model-registry.example.com:8443/models/15/versions/42",
+    "registrationTimestamp": "2024-01-15T10:30:00Z",
+    "registrationStatus": "success"
+}]
+```
+
+### Configuration Management
+
+Below are two configuration approaches, with Option 1 as the initial implementation and Option 2 as a potential future
+enhancement.
+
+#### Option 1: Extend the kfp-launcher ConfigMap
+
+Extend the existing `kfp-launcher` ConfigMap to include Model Registry configuration:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kfp-launcher
+  namespace: kubeflow
+data:
+  # Existing configuration...
+  pipelineRoot: "s3://kfp-artifacts"
+
+  # New Model Registry configuration
+  modelRegistry: |
+    url: https://model-registry.example.com:8443
+    tokenSecretRef: # Follows the same key names for existing configurations
+      secretName: model-registry-auth
+      secretNamespace: model-registry-system  # Defaults to current namespace
+      tokenKey: token
+    caConfigMapRef:  # Optional TLS certificate
+      configMapName: model-registry-ca-bundle
+      configMapNamespace: model-registry-system
+      key: ca-bundle.crt
+    timeout: 30s  # HTTP timeout for registration requests
+    retryAttempts: 3  # Number of retry attempts on failure
+```
+
+#### Option 2: Introduce LauncherConfig CRD
+
+Create a dedicated Custom Resource Definition for launcher configuration:
+
+```yaml
+apiVersion: pipelines.kubeflow.org/v2beta1
+kind: LauncherConfig
+metadata:
+  name: kfp-launcher
+spec:
+  modelRegistry:
+    url: https://model-registry.example.com:8443
+    tokenSecretRef:
+      name: model-registry-auth
+      namespace: model-registry-config # defaults to the current namespace
+      key: token
+    caConfigMapRef: # optional
+      name: model-registry-ca-bundle
+      namespace: model-registry-namespace # defaults to the current namespace
+      key: ca-bundle.crt
+    timeout: 30s
+    retryAttempts: 3
+```
+
+### User Stories
+
+#### Story 1
+
+As a data scientist, I would like to register my fine-tuned model to Model Registry in a pipeline without knowing the
+underlying Model Registry infrastructure and APIs, so that I can easily track model versions, share models with my team,
+and maintain a centralized model catalog as part of my automated ML workflows.
+
+#### Story 2
+
+As a data scientist, I would like to access a centralized model catalog that is independent of the specific tool used to
+create the models, enabling easy discovery and simplified model management across different ML workflows.
+
+### Notes/Constraints/Caveats
+
+N/A
+
+### Risks and Mitigations
+
+1. The Model Registry API doesn't have granular RBAC so the `pipeline-runner` service account has full access to the
+   Model Registry API.
+
+## Design Details
+
+### Architecture Overview
```

Comment:
This I'm not sure I fully understands: if you set a model registry client based on the configuration in the context of the KFP code, you can use the Model Registry python client directly: https://model-registry.readthedocs.io/en/latest/#registering-models

Below, it sounds to me the `.register()` is actually manually mapping to a request handled on the Go side.
I'm confused why this translation is required/preferred, can you expand on it please?

Reply:
it's indeed KFP choice, but the option I spoke earlier would give User the ability to perform further actions than what is currently designed here, for example remove labels from previous ModelVersions of a RegisteredModel, etc.

Reply:
> This way the user does not need to reach out to their admin for example, and ask for the URL and the Credentials/Access to the Model Registry they are writing to, we handle that all for them.

You could do similarly for having MR auth in every pipelinerun injected, by setting up some configuration with analogous mechanism by the Admin, and without having "user to reach out to their admin" :)

But regardless I see, you want to have an "_opinionated_ for simple Model tracking to MR" without exposing all the capabilities for now.
/lgtm

Reply:
@tarilabs we want to avoid injecting the credentials for security reasons to avoid leaking the credential to any pipeline. There are still ways for the pipeline to get it with this proposal since the Launcher runs with the same Kubernetes service account as the user's Python code, but it's something I want to fix later on. By going with an approach of injecting the credentials, it makes it harder to to close this gap.

We also don't want the user to being modifying other things in the Model Registry unrelated to the pipeline unless they provide their own credentials.

These are all things we could revisit later.